modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/distruptor/inbound/InboundTransactionEvent.java

```diff
@@ -85,6 +112,13 @@ public InboundTransactionEvent(MessageStore.AndesTransaction transaction, Inboun
      * @throws AndesException
      */
     public void commit() throws AndesException {
+
+        if(currentBatchSize > maxBatchSize) {
+            currentBatchSize = 0;
+            throw new AndesException("Current enqueued batch size exceeds maximum transactional batch size of " +
+            maxBatchSize + " bytes." );
+        }
+
         if (log.isDebugEnabled()) {
             log.debug("Prepare for commit");
```

Comment:
understood. I leave it to your judgement on having enough information to debug (at least channel ID)